core/src/main/java/bisq/core/payment/payload/RevolutAccountPayload.java

```diff
@@ -105,7 +110,34 @@ public static RevolutAccountPayload fromProto(protobuf.PaymentAccountPayload pro
 
     @Override
     public String getPaymentDetails() {
-        return Res.get(paymentMethodId) + " - " + Res.getWithCol("payment.account.userName") + " " + getUserName();
+        Tuple2<String, String> tuple = getLabelValueTuple();
+        return Res.get(paymentMethodId) + " - " + tuple.first + ": " + tuple.second;
+    }
+
+    private Tuple2<String, String> getLabelValueTuple() {
+        String label;
+        String value;
+        checkArgument(!userName.isEmpty() || hasOldAccountId(),
+                "Either username must be set or we have an old account with accountId");
+        if (!userName.isEmpty()) {
+            label = Res.get("payment.account.userName");
+            value = userName;
+
+            if (hasOldAccountId()) {
+                label += "/" + Res.get("payment.account.phoneNr");
+                value += "/" + accountId;
+            }
+        } else {
+            label = Res.get("payment.account.phoneNr");
+            value = accountId;
+        }
+        return new Tuple2<>(label, value);
+    }
+
+    public Tuple2<String, String> getRecipientsAccountData() {
+        Tuple2<String, String> tuple = getLabelValueTuple();
```

Comment:
Using `var tuple` really helps make the code cleaner and easier to read in cases like this. Although that's personal preference.

Reply:
I am not a fan of 'var', prefer to have the type information more explicit. I know in such a small context it does not make much difference, but in larger context it does.